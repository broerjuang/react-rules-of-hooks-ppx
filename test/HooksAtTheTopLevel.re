[@react.component]
let make = () => <div onClick={_evt => useMouseHook()} />;
