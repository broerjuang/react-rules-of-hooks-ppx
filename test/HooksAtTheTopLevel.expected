File "HooksAtTheTopLevel.ml", line 1, characters 42-57:
1 | let make () = ((div ~onClick:(fun _evt -> useMouseHook ()) ~children:[] ())
                                              ^^^^^^^^^^^^^^^
Error: Hooks can't be inside conditionals, neither loops.
